content/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure.md

```diff
@@ -0,0 +1,135 @@
+
+* Frederic Passeron gets assigned a studentid "student397" for workshop "API101".
+
+![Customers Table from Frontend DB server](/img/howto-wod-3.png "Customers Table from Frontend DB server")
+
+H﻿ere are the details of the participant info when registered to a given workshop.
+
+2﻿.A first email is sent to participant from the frontend server welcoming him to the workshop and informing him the deployment is ongoing and that a second email show be recieved shortly providing him with the necessary infos to logon  to the workshop environment.
+
+3﻿.The frontend server at the same time sends the necessary orders through a procmail API to the backend server. The mail sent to the backend server contains the following details :
+
+* Action Type ( CREATE, CLEANUP, RESET)
+* W﻿orkshop ID
+* S﻿tudent ID
+
+4﻿.The Backend Server recieves the order and processes it by  parsing the email recieved using the procmail API. the procmail API automates the managementof the workshops.
+
+Like any API, it uses verbs to perform tasks.
+
+* CREATE to deploy a workshop
+* C﻿LEANUP to delete a workshop
+* R﻿ESET to reset associated workshop's resource
+
+T﻿his means:
+
+* P﻿reparing any infrastructure that might be requiered for the workshop (Virtual Appliance, Virtual Machine, Docker Container, LDAP config, etc..)
+* G﻿enerate a random Password for the allocated student
+* D﻿eploy the workshop content on the jupyterhub server in the dedicated student home directory (Notebooks files necessary for the workshop)
+* Send back through API Calls to the frontend server the confirmation of the deployment of the workshop along with the student details (Password)
+
+5﻿.The frontend server get its different tables updated:
+
+* T﻿he customer tables shows an active status for the participant row. The password field has been updated. 
+* T﻿he Workshop table gets also updated. The capacity field decrement the number of available seats. 
+* The student tables gets updated as well by setting the allocated student to active.
+
+6﻿.The frontend server sends the second email to the particpant providing him with the details to connect to the workshop environment.
+
+### T﻿he Run Phase:
+
+F﻿rom the email, the particpant click on the start button. it will open up a browser to the jupyterhub server and directly open up the readme first notebook presenting the workshops flow.
+
+T﻿he participant will go through the different steps and labs of the workshop connecting to the necessary endpoints and leveraging the different kernels available on the jupyterhub server.
+
+M﻿eanwhile, the frontend server will perform regular checks on time passed. Depending on time allocation (from 2 to 4 hours) associated with the workshop, the frontend server will send a reminder email usually a hour before the end of the time allocated. The time count actually starts when the participant hit the register for the workshop button. It is mentionned in the terms and conditions.
```

Comment:
Add in "how much" and "has" to the end of the first sentence so it reads:
"Meanwhile, the frontend server will perform regular checks on how much time has passed."

Add "the" prior to "time allocation" in the second sentence. 

In the last sentence, change "the participant" to "participants".

"mentioned" has a typo.